openquake/hazardlib/gsim/mgmpe/stewart2020.py

```diff
@@ -0,0 +1,260 @@
+
+    out[vs30 < vw2] = w2
+
+    return out
+
+
+def _get_f760_model(C760, wimp):
+    # Implements eq. 5 in SeA2020
+    #
+    # param C760
+    #   Coefficients for
+    # param vs30
+    #   A 1D numpy array with vs30 values
+    # param wimp
+    #   A scalar or an array of values in [0, 1]. The array must have the
+    #   same cardinality of vs30
+
+    # Compute the weight for gradient
+    wgr = 1.0 - wimp
+
+    # Compute the scaling factor
+    return C760['f760i'] * wimp + C760['f760g'] * wgr
+
+
+def _get_vs30_scaling_model(C, vs30, f760):
+    # Implements eq. 3 in SeA2020
+    #
+    # param C
+    #   Coefficients for a given IMT from table COEFFS
+    # param vs30
+    #   A 1D numpy array with vs30 values
+    # param f760
+    #   The f760 scaling factors
+
+    # Initialise fv
+    fv = np.zeros(len(vs30))
+
+    # First interval - In the paper eq.3 they indicate vs30 > 200 but in figure
+    # 13 they show results for this vs30 value
+    idx = np.nonzero((vs30 >= CONSTS['vl']) & (vs30 <= C['v1']))[0]
+    if len(idx):
+        tmp = C['c'] * np.log(C['v1'] / CONSTS['vref'])
+        fv[idx] = tmp
+
+    # Second interval
+    idx = (vs30 > C['v1']) & (vs30 <= C['v2'])
+    if len(idx):
+        tmp = C['c'] * np.log(vs30[idx] / CONSTS['vref'])
+        fv[idx] = tmp
+
+    # Third interval
+    idx = (vs30 > C['v2']) & (vs30 <= CONSTS['vu'])
+    if len(idx):
+        fv[idx] = C['c'] * np.log(C['v2'] / CONSTS['vref'])
+
+    # Fourth interval
+    idx = (vs30 > CONSTS['vu']) & (vs30 <= 3000)
+    if len(idx):
+        term1 = C['c'] * np.log(C['v2'] / CONSTS['vref'])
+        term2 = C['c'] * np.log(C['v2'] / CONSTS['vref']) + f760[idx]
+        term3 = np.log(vs30[idx] / CONSTS['vu']) / np.log(3000.0 / CONSTS['vu'])
+        fv[idx] = term1 - term2 * term3
+    return fv
+
+
+def stewart2020_linear_scaling(imt, vs30, wimp=None, usgs=False):
+    """
+    Implements the Vs30 scaling model of Stewart et al. (2020; EQS).
+
+    NOTE a similar function is embedded in the NGA East implementation
+    `nga_east.py`.  For the time being we retain this implementation as this
+    code better fits the purpouse of being used within the modifiable GMPE.
+
+    :param imt:
+        Intensity measure type
+    :param vs30:
+        A vector with values of Vs30
+    :param wimp:
+        Weight for the impedance model (see eq. 5 of Stewart et al., 2020)
+    :param usgs:
+        Flag determining the use of the default coefficients or the ones
+        proposed by the USGS
+    :returns:
+        The natual logaritm of the ground-motion scaling due to the linear
+        component of amplification
+    """
+
+    # Check the Vs30 provided
+    if np.any(vs30 < CONSTS['vl']):
+        msg = 'Vs30 values are beyond the limits '
+        msg += ' admitted by the Stewart et al. (2020) model'
+        raise ValueError(msg)
+
+    # Get f760
+    if usgs:
+        import openquake.hazardlib.gsim.nga_east as ngae
```

Comment:
Why the import inside and not a top-level?

Reply:
as above